src/k8s-engine/wsl.ts

```diff
@@ -1207,8 +1230,25 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
             const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '')
               .replace('/var/log', logPath);
 
+            await this.writeFile('/etc/init.d/host-resolver', SERVICE_SCRIPT_HOST_RESOLVER, { permissions: 0o755 });
             await this.writeFile('/etc/init.d/dnsmasq-generate', SERVICE_SCRIPT_DNSMASQ_GENERATE, { permissions: 0o755 });
-            await this.execCommand('/sbin/rc-update', 'add', 'dnsmasq-generate', 'default');
+            await this.writeConf('host-resolver', {
+              RESOLVER_PEER_BINARY: await this.getHostResolverPeerPath(),
+              LOG_DIR:              logPath,
+            });
+            if (this.cfg?.experimentalHostResolver) {
+              console.debug(`launching experimental DNS host-resolver`);
+              try {
+                this.resolverHostProcess.start();
+              } catch (error) {
+                console.error('Failed to run host-resolver vsock-host process:', error);
+              }
+              await this.execCommand('/sbin/rc-update', 'add', 'host-resolver', 'default');
+              await this.execCommand('/sbin/rc-update', 'del', 'dnsmasq-generate', 'default');
```

Comment:
Forgot to mention… also needs `rc-update del dnsmasq default`. (`dnsmasq` and `dnsmasq-generate` are two distinct services.)